File: src/main/java/mods/eln/sixnode/signalinductor/SignalInductorDescriptor.java
package mods.eln.sixnode.signalinductor;

import mods.eln.node.six.SixNodeDescriptor;
import mods.eln.sim.mna.state.ElectricalLoad;
import mods.eln.sim.mna.passive.Inductor;
import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor;

public class SignalInductorDescriptor extends SixNodeDescriptor {

    ElectricalCableDescriptor cable;
    public double henri;

    public SignalInductorDescriptor(String name, double henri, ElectricalCableDescriptor cable) {
        super(name, SignalInductorElement.class, SignalInductorRender.class);
        this.henri = henri;
        this.cable = cable;
    }

    public void applyTo(ElectricalLoad load) {
        cable.applyTo(load);
    }

    public void applyTo(Inductor inductor) {
        inductor.setL(henri);
    }
}

File: src/main/kotlin/mods/eln/sixnode/signalinductor/SignalInductorDescriptor.kt
package mods.eln.sixnode.signalinductor

import mods.eln.node.six.SixNodeDescriptor
import mods.eln.sim.ElectricalLoad
import mods.eln.sim.mna.component.Inductor
import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor
import net.minecraft.item.Item

class SignalInductorDescriptor(name: String?, var henri: Double, @JvmField var cable: ElectricalCableDescriptor) :
    SixNodeDescriptor(name, SignalInductorElement::class.java, SignalInductorRender::class.java) {
    var descriptor: String? = null

    override fun setParent(item: Item, damage: Int) {
        super.setParent(item, damage)
    }

    fun applyTo(load: ElectricalLoad?) {
        cable.applyTo(load)
    }

    fun applyTo(inductor: Inductor) {
        inductor.inductance = henri
    }
}
